app/models/service_provider.rb

```diff
@@ -51,9 +52,12 @@ def skip_encryption_allowed
 
   private
 
+  # @return [String,nil]
   def load_cert(cert)
-    cert_file = Rails.root.join('certs', 'sp', "#{cert}.crt")
-    return OpenSSL::X509::Certificate.new(cert) unless File.exist?(cert_file)
-    File.read(cert_file)
+    if cert.include?('-----BEGIN CERTIFICATE-----')
+      cert
+    elsif (cert_file = Rails.root.join('certs', 'sp', "#{cert}.crt")) && File.exist?(cert_file)
```

Comment:
I did this check first because if something looks like cert content, we don't want to do an exist check on it, the filesystem will barf

Some prior art in the Dashboard: https://github.com/18F/identity-dashboard/blob/838daaf8a0b470823b400ad1838af6e5da468f23/app/models/service_provider.rb#L136-L147

Reply:
**praise:** nice!